src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -117,6 +117,11 @@ internal bool TryExport(T data)
             // either the queue is full or exceeded the spin limit, drop the item on the floor
             Interlocked.Increment(ref this.droppedCount);
 
+            if (Sdk.CircularBufferDroppedCount.Enabled)
+            {
+                Sdk.CircularBufferDroppedCount.Add(1);
```

Comment:
callback is best suited here, as we are already calculating droppedCount, and no need to force Metrics SDK to redo the work.